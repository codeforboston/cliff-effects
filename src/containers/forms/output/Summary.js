import { connect } from 'react-redux';

import { Summary } from '../../../forms/output/Summary';

const mapStateToProps  = (state) => {
  return { client: state.get('client') };
};

export default connect(mapStateToProps)(Summary);
