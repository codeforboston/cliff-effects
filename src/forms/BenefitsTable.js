// REACT COMPONENTS
import React from 'react';
import {
  Table
} from 'semantic-ui-react';

// CUSTOM COMPONENTS
// Both the table and graph should just be added to a results page, but
// this will do for now

// BENEFIT LOGIC
import { getSNAPBenefits } from '../programs/federal/snap';
import { getHousingBenefit } from '../programs/state/massachusetts/housing';

// OBJECT MANIPULATION
import { cloneDeep } from 'lodash';


const getSignSymbol = function ( num ) {
  if ( num > 0 ) { return '+'; }
  else if ( num < 0 ) { return '-'; }
  else { return ''; }
};  // End getSignSymbol()


const BenefitsTable = function ( props ) {

  var currentClient = cloneDeep( props.client ),
      futureClient  = cloneDeep( currentClient );
  // Hack because everything's using future value instead
  // of just using current value.
  currentClient.future.earned = currentClient.current.earned;




  var SNAPBenefitCurrent  = Math.round( getSNAPBenefits( currentClient ).benefitValue * 12 ),
      SNAPBenefitFuture   = Math.round( getSNAPBenefits( futureClient ).benefitValue * 12 ),
      SNAPDiff            = SNAPBenefitFuture - SNAPBenefitCurrent,
      sec8BenefitCurrent  = Math.round( getHousingBenefit( currentClient ).benefitValue * 12 ),
      sec8BenefitFuture   = Math.round( getHousingBenefit( futureClient ).benefitValue * 12 ),
      sec8Diff            = sec8BenefitFuture - sec8BenefitCurrent,
      totalBenefitCurrent = SNAPBenefitCurrent + sec8BenefitCurrent,
      totalBenefitFuture  = SNAPBenefitFuture + sec8BenefitFuture,
      totalDiff           = SNAPDiff + sec8Diff,
      incomeCurrent       = Math.round( currentClient.current.earned * 12 ),
      incomeFuture        = Math.round( futureClient.future.earned * 12 ),
      incomeDiff          = incomeFuture - incomeCurrent,
      netCurrent          = totalBenefitCurrent + incomeCurrent,
      netFuture           = totalBenefitFuture + incomeFuture,
      netDiff             = totalDiff + incomeDiff;


const   columnHeaderStyle = {
                            background: 'rgba(0, 181, 173, 1)',
                            color: 'white',
                            fontSize: '1.3em',
                            fontWeight: 900,
                            textAlign: 'center',
                            borderRadius: 'inherit',
                            letterSpacing: '0.02em',
                            }
      , totalsRowStyle    = {
                            borderTop: "2px solid rgba(0, 181, 173, 1)",
                            fontWeight: 700,
                            fontSize: '1.1em',
                            padingTop: '0.25em'
                              }
      , rowHeaderStyle    = {
                    fontSize: '1.1em',
                    fontWeight: 700,
                    textAlign: 'left'
                    }
      , totalsRowHeaderStyle = {
                    fontSize: '1.2em',
                    fontWeight: 700,
                    textAlign: 'left',
                    borderTop: "2px solid rgba(0, 181, 173, 1)",
                    padingTop: '0.25em'


      };






const SNAPBenefitRow = function( props ){
  if(!currentClient.current.hasSnap ) return (null)
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>SNAP</Table.Cell>
        <Table.Cell textAlign='right'>${SNAPBenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${SNAPBenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(SNAPDiff) } ${Math.abs(SNAPDiff)} / year</Table.Cell>
      </Table.Row>
    )
  };

const Sec8BenefitRow  = function( props ){  
  if(!currentClient.current.hasHousing) return (null)
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>Section 8 Housing</Table.Cell>
        <Table.Cell textAlign='right'>${sec8BenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${sec8BenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(sec8Diff) } ${Math.abs(sec8Diff)} / year</Table.Cell>
      </Table.Row>
    ) 
};

const TotalBenefitsRow = function( props ){
  if(!currentClient.current.hasSnap || !currentClient.current.hasHousing) return (null)
    return(
      <Table.Row>
        <Table.Cell textAlign='right' width={3} style={totalsRowHeaderStyle}>Total Benefits</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${totalBenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${totalBenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>{ getSignSymbol(totalDiff) } ${Math.abs(totalDiff)} / year</Table.Cell>
      </Table.Row>
    )
};

const IncomeRow = function ( props ) { 
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>Income</Table.Cell>
        <Table.Cell textAlign='right'>${incomeCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${incomeFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(incomeDiff) } ${Math.abs(incomeDiff)} / year</Table.Cell>
      </Table.Row>
    )
};

const TotalsRow = function ( props ) {
    return (
      <Table.Row style={{border: 'none'}}>
        <Table.Cell textAlign='right' width={3} style={totalsRowHeaderStyle}>Net Total</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${netCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${netFuture} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>{ getSignSymbol(netDiff) } ${Math.abs(netDiff)} / year</Table.Cell>
      </Table.Row>
    )
  };


  return (
    <wrapper>
      <Table celled>
       <Table.Header>
          <Table.Row >
            <Table.Cell style={columnHeaderStyle} width={3}>Benefit</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>Current Benefits</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>New Estimate</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>Difference</Table.Cell>
          </Table.Row>
      </Table.Header>
        <Table.Body>
          <SNAPBenefitRow client={currentClient} />
          <Sec8BenefitRow client={currentClient} />
          <TotalBenefitsRow client={currentClient} />
          <IncomeRow />
          <TotalsRow />
        </Table.Body>
      </Table>
    </wrapper>
  )

};  // End BenefitsTable(<>)


export {
	BenefitsTable
};
