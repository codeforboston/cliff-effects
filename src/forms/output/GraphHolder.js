import React, { Component } from 'react';

// CUSTOM COMPONENTS
import { GraphTimeButtons } from '../../components/GraphTimeButtons';


class GraphHolder extends Component {

  constructor (props) {
    super(props);
    this.state = { activeID: 'Yearly' };
  }

  onClick = (evnt) => {
    let id = evnt.target.id;
    this.setState({ activeID: id });
  };

  render () {
    const { activeID }  = this.state,
          { Graph, client }         = this.props,
          { current }               = client,
          // The ids later used to access all program-specific data and functions
          // Only active programs are added
          activePrograms            = [ ...current.benefits ];

    return (
      <div className='graph-holder'>
        <Graph
          className='client-graph'
          client={ client }
          timescale={ activeID }
          activePrograms={ activePrograms } />
        <GraphTimeButtons
          activeID={ activeID }
          onClick={ this.onClick } />
      </div>
    );
  };  // End render()

};  // End <GraphHolder>


export { GraphHolder };
