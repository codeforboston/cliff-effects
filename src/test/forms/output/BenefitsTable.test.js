import React from 'react';
import renderer from 'react-test-renderer';
import { cloneDeep, set } from 'lodash';

import { BenefitsTable } from '../../../forms/output/BenefitsTable';
import { snippets } from '../../helpers';
import { CLIENT_DEFAULTS } from '../../../utils/CLIENT_DEFAULTS';

const buildSnapshot = (client) => {
  const rendered = renderer.create(
    <BenefitsTable 
      client={ client } 
      snippets={ snippets } />
  );

  return rendered.toJSON();
};

test('Benefits table renders correctly', () => {
  const client = cloneDeep(CLIENT_DEFAULTS);
  expect(buildSnapshot(client)).toMatchSnapshot();

  const benefits = [ 'snap' ];

  set(client, 'current.benefits', benefits);
  expect(buildSnapshot(client)).toMatchSnapshot();

  set(client, 'current.earned', 100);
  expect(buildSnapshot(client)).toMatchSnapshot();

  set(client, 'future.earned', 200);
  expect(buildSnapshot(client)).toMatchSnapshot();

  set(client, 'current.earned', 300);
  expect(buildSnapshot(client)).toMatchSnapshot();

  set(client, 'future.earned', 400);
  expect(buildSnapshot(client)).toMatchSnapshot();
});
