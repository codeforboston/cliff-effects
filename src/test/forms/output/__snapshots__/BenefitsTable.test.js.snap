// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefits table renders correctly 1`] = `
<div>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          0
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          0
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          
           
          Some translated text.
          0
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
        style={
          Object {
            "border": "none",
          }
        }
      >
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.2em",
              "fontWeight": 700,
              "padingTop": "0.25em",
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          0
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          0
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          
           
          Some translated text.
          0
          Some translated text.
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Benefits table renders correctly 2`] = `
<div>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          
           
          Some translated text.
          0
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          0
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          0
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          
           
          Some translated text.
          0
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
        style={
          Object {
            "border": "none",
          }
        }
      >
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.2em",
              "fontWeight": 700,
              "padingTop": "0.25em",
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          
           
          Some translated text.
          0
          Some translated text.
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Benefits table renders correctly 3`] = `
<div>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          
           
          Some translated text.
          0
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          100
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          0
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          -
           
          Some translated text.
          100
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
        style={
          Object {
            "border": "none",
          }
        }
      >
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.2em",
              "fontWeight": 700,
              "padingTop": "0.25em",
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          292
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          -
           
          Some translated text.
          100
          Some translated text.
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Benefits table renders correctly 4`] = `
<div>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          
           
          Some translated text.
          0
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          100
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          200
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          +
           
          Some translated text.
          100
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
        style={
          Object {
            "border": "none",
          }
        }
      >
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.2em",
              "fontWeight": 700,
              "padingTop": "0.25em",
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          292
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          392
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          +
           
          Some translated text.
          100
          Some translated text.
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Benefits table renders correctly 5`] = `
<div>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          
           
          Some translated text.
          0
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          300
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          200
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          -
           
          Some translated text.
          100
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
        style={
          Object {
            "border": "none",
          }
        }
      >
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.2em",
              "fontWeight": 700,
              "padingTop": "0.25em",
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          492
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          392
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          -
           
          Some translated text.
          100
          Some translated text.
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Benefits table renders correctly 6`] = `
<div>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="three wide"
          style={
            Object {
              "background": "rgba(0, 181, 173, 1)",
              "borderRadius": "inherit",
              "color": "white",
              "fontSize": "1.3em",
              "fontWeight": 900,
              "letterSpacing": "0.02em",
              "textAlign": "center",
            }
          }
        >
          Some translated text.
        </td>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          192
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          187
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          -
           
          Some translated text.
          5
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "fontSize": "1.1em",
              "fontWeight": 700,
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          300
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          Some translated text.
          400
          Some translated text.
        </td>
        <td
          className="right aligned"
        >
          +
           
          Some translated text.
          100
          Some translated text.
        </td>
      </tr>
      <tr
        className=""
        style={
          Object {
            "border": "none",
          }
        }
      >
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.2em",
              "fontWeight": 700,
              "padingTop": "0.25em",
              "textAlign": "left",
            }
          }
        >
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          492
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          Some translated text.
          587
          Some translated text.
        </td>
        <td
          className="right aligned three wide"
          style={
            Object {
              "borderTop": "2px solid rgba(0, 181, 173, 1)",
              "fontSize": "1.1em",
              "fontWeight": 700,
              "padingTop": "0.25em",
            }
          }
        >
          +
           
          Some translated text.
          95
          Some translated text.
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
